src/main/java/microsoft/sql/Types.java

```diff
@@ -74,4 +74,10 @@ private Types() {
      * Microsoft SQL type GEOGRAPHY.
      */
     public static final int GEOGRAPHY = -158;
+    
+    /**
+     * The constant in the Java programming language, sometimes referred to as a type code, that identifies the
+     * Microsoft SQL type JSON.
+     */
+    public static final int JSON = -159;
```

Comment:
just wondering, where does this come from? I looked into this before but couldnt find any reference

Reply:
This class (ie. microsoft.sql.Types.java) defines constants that are used to identify SQL types specific to Microsoft SQL Server, which are not defined in java.sql.Types. The value -159 is simply the next sequential number in our microsoft.sql.Types.java.
 
public class MicrosoftSqlTypes {

    /**
     * The constant in the Java programming language, sometimes referred to as a type code, that identifies the
     * Microsoft SQL type GEOGRAPHY.
     */
    public static final int GEOGRAPHY = **-158;**

    /**
     * The constant in the Java programming language, sometimes referred to as a type code, that identifies the
     * Microsoft SQL type JSON.
     */
    public static final int JSON = **-159;**
}